ecs-init/cache/dependencies.go

```diff
@@ -141,6 +141,13 @@ type instanceMetadata interface {
 	Region() (string, error)
 }
 
+type blackholeInstanceMetadata struct {
+}
+
+func (b *blackholeInstanceMetadata) Region() (string, error) {
+	return "", errors.New("blackholed")
```

Comment:
why should this return an error? Might this cause any issue down the funnel?

Reply:
if we don't return error here, the empty string will be treated as a valid region, which i think will break the logic of using it later here https://github.com/aws/amazon-ecs-init/blob/b5d2758a0ba113cd06d5e9867e15c49c206d3207/ecs-init/cache/cache.go#L183